File: src/lib/types/k8s/port-forwarding.ts
// Port forwarding types
export interface PortForward {
  id: string;
  namespace: string;
  resourceName: string;
  resourceType: string;
  localPort: number;
  remotePort: number;
  status: string;
  createdAt: string;
  url?: string;
}

export interface PortForwardOptions {
  localPort?: number;
  remotePort: number;
}

File: src-tauri/src/k8s/types/port_forwarding.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Serialize, Deserialize, Clone)]
pub struct PortForward {
    pub id: String,
    pub namespace: String,
    pub resource_name: String,
    pub resource_type: String,
    pub local_port: i32,
    pub remote_port: i32,
    pub status: String,
    pub created_at: String,
    pub url: Option<String>,
}
